### DISCUSSION TITLE
[✨] Ability to handle "no internet" mode

### DISCUSSION THREAD

**Discussion by shairez** [QwikDev/qwik-evolution#181] [Proposals For Qwik]
### Is your feature request related to a problem?

On a mobile device, when users go offline randomly, prefetching might not work

Then a user can click a button, and just see no response. 

### Describe the solution you'd like

Idea: a way for either the Qwik Loader or some other plugin that get prefetched first to let the user know that code is not loaded because they are offline. 

So once a user clicks the button, they see a message (which could be a generic one customized by the app's devs) that interactivity is not possible because they are offline. 

This is still a rough idea, but I think it's going to be a common use case where prefetching might fail and users won't know why the app looks interactive but it's not. 

### Describe alternatives you've considered

1. Qwik loader hook
2. Another plugin that gets prefetched by default after the html loads (though it might suffer from the same problem ironically)

**Comment by manucorporat** [comment]
it could be implemented by qwikloader emiting an event, that can be captured by other code to handle it:
https://github.com/BuilderIO/qwik/blob/main/packages/qwik/src/qwikloader.ts

**Comment by wmertens** [comment]
How about a fallback in case there's no event handler, that just displays an alert?

**Comment by PatrickJS** [comment]
this is similar to trying to handle skew issue so I'll assign it to myself to figure it out

**Comment by genki** [comment]
I think there should be a handler for fail of the dynamic import in the qwikloader.

The handler should do
* handle the error and enqueue the failed import.
* listen to the "online" event of the window object in order to re-import the failed imports.
* resolve the promise of the import after the success of the re-import.

The developers should suppose

* every execution of the QRL can fail when the network is offline.
* to provide error message for that
* or to wait for the network is online again and the QRL being re-executed.

So the QRL may have information to determine that the execution would fail on offline or wait for online.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Reply by berli888** [reply]
Hi guys :)
do we have any update on this feature ?
Is there any way right now to tell the user that the application is broken because we are in offline mode ?
